### ISSUE TITLE
Document spooling and infinite retry settings for additional reliability with monitoring data

### ISSUE THREAD

**Issue by dedemorton** [elastic/beats#15067]
This issue came up in https://github.com/elastic/enhancements/issues/7134#issuecomment-562111671

>We can document (e.g. after step 5. over here https://www.elastic.co/guide/en/beats/filebeat/current/monitoring-metricbeat-collection.html#configure-metricbeat), the addition of spooling and infinite retry settings, for users who the additional reliability with their monitoring data. To be clear, this would be an additional "If you want..." section at the end of step 5, not part of the default configuration we are showing in step 5. right now.
